Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/TestFixture_02_InteractionDefinitionTests.cs

```diff
@@ -234,6 +235,13 @@ public void Test07_TestVector2Changed()
 
             Assert.True(setValue2 == testValue2);
             Assert.IsFalse(interaction.Changed);
+
+            // Make sure changes to a single component are detected correctly when only one axis is inverted (bug #4114)
+            interaction.InvertXAxis = true;
+            interaction.InvertYAxis = false;
+            interaction.Vector2Data = Vector2.one;
+            interaction.Vector2Data = new Vector2(1, -1);
+            Assert.IsTrue(interaction.Changed);
```

Comment:
Could we format this test like the others above where we're also testing the return value?

Typically it's best to also test setting the value with the same value again to ensure that not change has happened as well.

Thanks :)

Reply:
> Could we format this test like the others above where we're also testing the return value?

I find this formatting quite natural and clear. Can you paste the test formatted as you'd like to see it?

Reply:
Sure take a look [here](https://github.com/XRTK/XRTK-Core/pull/138/files)

I added an additional test case that was specifically designed to test the inverted axis types, and updated the tests to also validate two additional value changes. 

We're testing:

1. set initial value
2. did value change? `Asset.False`
3. set to value 1
4. did value change? `Assert.True`
5. set value 1 again
6. did value change? `Asset.False`
7. repeat for each additional value change (we used 4 additional values)
8. set initial value
9. did the value change? `Assert.True`
10. set initial value again
11. did the value change? `Assert.False`